crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -141,29 +141,57 @@ impl ReadView {
     pub fn transaction(&self, tx_id: &TxId) -> StorageResult<Transaction> {
         let result = self.on_chain.transaction(tx_id);
         if result.is_not_found() {
-            if let Some(tx) = self.off_chain.old_transaction(tx_id)? {
-                Ok(tx)
-            } else {
-                Err(not_found!(Transactions))
-            }
+            self.off_chain.old_transaction(tx_id)
         } else {
             result
         }
     }
 
-    pub async fn transactions(
+    pub async fn transactions(&self, tx_ids: &[TxId]) -> Vec<StorageResult<Transaction>> {
+        let transactions: Vec<_> = self
+            .on_chain
+            .transactions(tx_ids.iter().into_boxed())
+            .collect();
```

Comment:
I suspect you can avoid collecting here since later you iterate on transactions again.

Reply:
Unforutnately I need to both iterate over the transactions and zip them with the on_chain_results - so we need to iterate twice over the tersults. We could potentially do something fancy with [Itertools::tee](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.tee), but I think this solution is simpler and easier on the eyes.